netbeans/CheckerFrameworkPlugin/src/org/checkerframework/netbeans/CheckerFrameworkPanel.java

```diff
@@ -0,0 +1,159 @@
+package org.checkerframework.netbeans;
+
+import java.awt.event.ItemEvent;
+import java.awt.event.ItemListener;
+import java.io.IOException;
+import java.io.InputStream;
+import java.util.Map.Entry;
+import java.util.Set;
+import javax.swing.BoxLayout;
+import javax.swing.JCheckBox;
+import javax.swing.JLabel;
+import javax.swing.JPanel;
+import org.netbeans.api.project.ProjectManager;
+import org.netbeans.spi.project.support.ant.EditableProperties;
+import org.openide.filesystems.FileObject;
+import org.openide.filesystems.FileUtil;
+import org.openide.modules.InstalledFileLocator;
+import org.openide.util.MutexException;
+
+public class CheckerFrameworkPanel extends JPanel {
+
+    private EditableProperties editableProperty;
```

Comment:
You should be able to make this field final. You need to think about what a good value is in case the properties couldn't be loaded. For checkerStrings you use an empty hashmap. Maybe the same will work here?